example/js/high-level.mjs

```diff
@@ -1,58 +1,65 @@
 import wasm from "./wasm.mjs"
 import * as diplomatRuntime from "./diplomat-runtime.mjs"
 
-const ICU4XFixedDecimalFormat_destroy_registry = new FinalizationRegistry(underlying => {
-  wasm.ICU4XFixedDecimalFormat_destroy(underlying);
+const ICU4XLocale_destroy_registry = new FinalizationRegistry(underlying => {
+  wasm.ICU4XLocale_destroy(underlying);
 });
 
-export class ICU4XFixedDecimalFormat {
+const ICU4XLocale_value_destroy_registry = new FinalizationRegistry(obj => {
+  wasm.ICU4XLocale_drop_ptr(obj["ptr"]);
+  wasm.diplomat_free(obj["ptr"], obj["size"]);
+});
+
+export class ICU4XLocale {
   constructor(underlying) {
     this.underlying = underlying;
   }
 
-  static new(locale, provider) {
+  static new(name) {
+    let name_diplomat_bytes = (new TextEncoder()).encode(name);
+    let name_diplomat_ptr = wasm.diplomat_alloc(name_diplomat_bytes.length);
+    let name_diplomat_buf = new Uint8Array(wasm.memory.buffer, name_diplomat_ptr, name_diplomat_bytes.length);
+    name_diplomat_buf.set(name_diplomat_bytes, 0);
     const diplomat_out = (() => {
-      const out = new ICU4XFixedDecimalFormat(wasm.ICU4XFixedDecimalFormat_new(locale.underlying, provider.underlying));
-      ICU4XFixedDecimalFormat_destroy_registry.register(out, out.underlying);
+      const out = new ICU4XLocale(wasm.ICU4XLocale_new(name_diplomat_ptr, name_diplomat_bytes.length));
+      ICU4XLocale_destroy_registry.register(out, out.underlying);
       return out;
     })();
-    return diplomat_out;
-  }
-
-  format_write(value) {
-    const diplomat_out = diplomatRuntime.withWriteable(wasm, (writeable) => wasm.ICU4XFixedDecimalFormat_format_write(this.underlying, value.underlying, writeable));
+    wasm.diplomat_free(name_diplomat_ptr, name_diplomat_bytes.length);
     return diplomat_out;
   }
 }
 
-const ICU4XFixedDecimal_destroy_registry = new FinalizationRegistry(underlying => {
-  wasm.ICU4XFixedDecimal_destroy(underlying);
+const ICU4XFixedDecimalFormat_destroy_registry = new FinalizationRegistry(underlying => {
+  wasm.ICU4XFixedDecimalFormat_destroy(underlying);
 });
 
-export class ICU4XFixedDecimal {
+const ICU4XFixedDecimalFormat_value_destroy_registry = new FinalizationRegistry(obj => {
+  wasm.ICU4XFixedDecimalFormat_drop_ptr(obj["ptr"]);
+  wasm.diplomat_free(obj["ptr"], obj["size"]);
+});
+
+export class ICU4XFixedDecimalFormat {
   constructor(underlying) {
     this.underlying = underlying;
   }
 
-  static new(v) {
+  static try_new(locale, provider) {
```

Comment:
issue: I think in JS we should just be throwing in cases like this. But while we're waiting on Result types I guess this suffices.

Reply:
#16